syntax = 'proto3';

package web_nst;
enum ImageType {
  CONTENT = 0;
  STYLE = 1;
}

message GiveJobRequest{
  string workerId = 1;
}
message GiveJobResponse{
  bytes contentImage = 1;
  bytes styleImage = 2;
  string requestId = 3;
  string workerId = 4;
  bool jobGiven = 5;
}

message JobProgressData{
  string workerId = 1 ;
  string requestId = 2;
  float progressPercent = 3;
}
message Ok {
}
message JobEndRequest {
  bytes image = 1;
  string requestId = 2;
  string workerId = 3;
}


service JobService {
  rpc GiveJob(GiveJobRequest) returns (GiveJobResponse);
  rpc JobProgress(stream JobProgressData) returns (Ok);
  rpc JobEnd(JobEndRequest) returns (Ok);
}